File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
import numpy as np

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums)==0:
            return 0
        aa = np.sort(nums)
        nums1 = aa[:-1]
        nums2 = aa[1:]
        #diff = np.array(nums2) - np.array(nums1)
        mm = 0
        mmm = 0
        for ii in range(len(nums2)):
            if (nums2[ii]-nums1[ii]==1):
                mm = mm+1
            elif (nums2[ii]-nums1[ii]==0):
                continue
            else:
                mm = 0
            print(mm)
            if mm > mmm:
                mmm = mm
        return mmm+1

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        set<int> sortedNums;
        for (int i=0; i<nums.size(); i++){
            sortedNums.insert(nums[i]);
        }       
        int cnt = 0;
        int mcnt = 0;
        set<int>::iterator it;
        int preval = 0;
        
        //[]비어있을 경우
        if (nums.empty()){
            return (int)0;}
        
        for (it=sortedNums.begin(); it!=sortedNums.end(); it++){
            //첫번째 값의 경우
            if (it==sortedNums.begin()){
                preval = *it;
                mcnt = cnt;
            }
            //2번째 값부터 앞의 값과 비교
            else{
                if (*it==preval+1){

                    cnt++;
                    if (mcnt<cnt)
                        {mcnt = cnt;}
                }
                else{
                    cnt = 0;
                }
                preval = *it;
            }  
        }
        mcnt++;
        return mcnt;
    }
};